flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/KinesisStreamShard.java

```diff
@@ -61,19 +68,23 @@ public String getShardId() {
 	}
 
 	public String getStartingSequenceNumber() {
-		return shard.getSequenceNumberRange().getStartingSequenceNumber();
+		SequenceNumberRange sequenceNumberRange = shard.getSequenceNumberRange();
+		return (sequenceNumberRange == null) ? null : sequenceNumberRange.getStartingSequenceNumber();
```

Comment:
The `Shard`s returned from actual Kinesis will never have null values for these.